Rv/feature ci

FWBlueWizard opened the pull request:
Adding CircleCI updates to Optimism fork. 

* Support RV kontrol Workflow executions of kontrol on changes to kontrol folder 

This PR Supports Optimism Engagement [PR](https://github.com/ethereum-optimism/optimism/pull/8271)

JuanCoRo committed:
Add dummy verification

JuanCoRo committed:
Add logging to execution script

JuanCoRo committed:
Add `Counter.sol` file to `bedrock/src/L1` folder

JuanCoRo committed:
Export `FOUNDRY_PROFILE` variable

FWBlueWizard committed:
Updatin circleci config to run kontrol tests when changes occur in kontrol folder, top-level

JuanCoRo commented on packages/contracts-bedrock/src/L1/Counter.sol:
```diff
@@ -0,0 +1,14 @@
+    function setNumber(uint256 newNumber) public {
+        number = newNumber;
+    }
+
+    function increment() public {
+        number++;
+    }
```
```suggestion
```
This function is not actually used

JuanCoRo commented on packages/contracts-bedrock/test/kontrol/Counter.t.sol:
```diff
@@ -0,0 +1,23 @@
+
+/* import {Test} from "forge-std/Test.sol"; */
+import {Counter} from "src/L1/Counter.sol";
+import {OptimismPortal} from "src/L1/OptimismPortal.sol";
```
```suggestion
```

JuanCoRo commented on packages/contracts-bedrock/test/kontrol/Counter.t.sol:
```diff
@@ -0,0 +1,23 @@
+contract CounterTest {
+
+    Counter counter;
+    OptimismPortal optimismPortal;
```
```suggestion
```

JuanCoRo commented on packages/contracts-bedrock/test/kontrol/Counter.t.sol:
```diff
@@ -0,0 +1,23 @@
+
+    function setUp() public {
+        counter = new Counter();
+        optimismPortal = new OptimismPortal();
```
```suggestion
```

JuanCoRo commented on .circleci/config.yml:
```diff
@@ -1269,6 +1269,48 @@ jobs:
+    when:
+      changes:
+        paths:
+          - packages/contracts-bedrock/test/kontrol/**
```
```suggestion
          - packages/contracts-bedrock/test/kontrol/**
          - packages/contracts-bedrock/src/L1/OptimismPortal.sol
```
Adding `OptimismPortal.sol` so that it triggers on changes to that file

JuanCoRo committed:
Update packages/contracts-bedrock/src/L1/Counter.sol

JuanCoRo committed:
Update packages/contracts-bedrock/test/kontrol/Counter.t.sol

JuanCoRo committed:
Update packages/contracts-bedrock/test/kontrol/Counter.t.sol

JuanCoRo committed:
Update packages/contracts-bedrock/test/kontrol/Counter.t.sol

JuanCoRo committed:
Update .circleci/config.yml

JuanCoRo committed:
Move `Counter.sol` to Kontrol folder

JuanCoRo committed:
Remove redundant remappings

JuanCoRo committed:
Change `when` key in favor of `check-changed`

JuanCoRo committed:
Add `kout` to .gitignore

JuanCoRo committed:
Add dedicated log folder for Kontrol

JuanCoRo committed:
Explicit declaration for `machine: true`

JuanCoRo committed:
Remove `Checkout submodules` in favor of `forge build`

`git submodule sync` and `git submodule update --init --recursive` should be subsumed by `forge build`, which is run when the execution script is triggered. Hence the removal of this step

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

JuanCoRo committed:
Format `Counter.t.sol` properly

JuanCoRo committed:
Merge remote-tracking branch 'origin/develop' into rv/feature-ci

JuanCoRo committed:
Remove `when: always` key

JuanCoRo committed:
Add `.kontrolrc` file

FWBlueWizard committed:
Update config.yml

FWBlueWizard committed:
Moving run steps to a script

FWBlueWizard committed:
Update permissions, Add comments to script

FWBlueWizard committed:
Use Kontrol image for base image

FWBlueWizard committed:
Typo in image name

FWBlueWizard committed:
Bad end quote in script

JuanCoRo committed:
Set `kontrol-tests` resource class to `xlarge`

JuanCoRo committed:
`run-kontrol.sh`: Add description of setting-unsetting pattern

JuanCoRo committed:
Merge remote-tracking branch 'origin/develop' into rv/feature-ci

FWBlueWizard committed:
Need to point to the proper directory where the test are run and the kout folder is generated relative to repository top level

JuanCoRo committed:
Try `large` containers for `kontrol-tests`

JuanCoRo committed:
Back to xlarge

FWBlueWizard committed:
Saving testable scripts and pnpm package.json additions

FWBlueWizard committed:
Script section name change

FWBlueWizard committed:
Fixing path, dirname stripped a necessary folder

FWBlueWizard committed:
Updating job to install docker-cli to ci image for calling and executing the kontrol image from remote

FWBlueWizard committed:
Remove version from script input, use kontrolrc file

FWBlueWizard committed:
wrong path to kontrorc file. Remove version from circle config

FWBlueWizard committed:
Removing old script, consolidated to run-kontrol.sh

FWBlueWizard committed:
Fixing missing failure

FWBlueWizard committed:
update loggint o publish results on pass AND FAIL

FWBlueWizard committed:
Add back docker stop

FWBlueWizard committed:
Missing an environment varibale for foundry profile in container

FWBlueWizard committed:
Isolate docker cleanup function, retesting previous failure

FWBlueWizard committed:
Recursively check out bedrock submodules

FWBlueWizard committed:
Creating tar was not necessary, removing.

FWBlueWizard committed:
Updating artifact upload reference path, and destination name

FWBlueWizard committed:
Create 2 output files from results, 1 with static name.

JuanCoRo committed:
Update packages/contracts-bedrock/test/kontrol/kontrol/run-kontrol.sh

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

FWBlueWizard committed:
Fixing shellcheck analysis

FWBlueWizard committed:
Per https://github.com/foundry-rs/foundry/issues/6519, run update first

FWBlueWizard committed:
Instll as well to run git submodule update

FWBlueWizard committed:
Reverting to git direct method of checkout

FWBlueWizard committed:
Step description

FWBlueWizard committed:
Updating to fix most shellcheck findings. Kontrol commands do not run properly when the variable is quoted

FWBlueWizard committed:
Updating to fix most shellcheck findings. Kontrol commands do not run properly when the variable is quoted

JuanCoRo committed:
Merge remote-tracking branch 'origin/develop' into rv/feature-ci
